hive_generator/lib/src/class_builder.dart

```diff
@@ -93,7 +93,7 @@ class ClassBuilder extends Builder {
   String _cast(DartType type, String variable) {
     var suffix = _suffixFromType(type);
     if (hiveListChecker.isAssignableFromType(type)) {
-      return '($variable as HiveList$suffix)?.castHiveList()';
+      return '($variable as HiveList$suffix).castHiveList()';
```

Comment:
I think this might cause runtime error for nullable HiveList because the value might be null.

```suggestion
      return '($variable as HiveList$suffix)$suffix.castHiveList()';
```

This will make sure nullable values will be casted as `... as HiveList?)?.castHiveList()` and not nullable values will be casted using `... as HiveList).castHiveList()`.